lib/puppet/provider/datacat_collector/datacat_collector.rb

```diff
@@ -9,6 +9,16 @@ def exists?
       r.is_a?(Puppet::Type.type(:datacat_fragment)) && ((our_names & [ r[:target] ].flatten).size > 0)
     end
 
+    # decrypt any encrypted fragments
+    if defined?(Puppet_X::Binford2k::NodeEncrypt)
+      fragments.each do |fragment|
+        fragment[:data].each do |key,value|
+          next unless Puppet_X::Binford2k::NodeEncrypt.encrypted?(value)
```

Comment:
I'm not keen on this particular check as it smells like it's checking for a magic number, and it also seems too soft. I think I'd sooner have an additional explicit flag on the fragment, something more like:

```
fragments.each do |fragment|
 if fragment[:encrypted]
   fragment[:data].each do |key,value|
     fragment[:data][key] = Puppet_X::Binford2k::NodeEncrypt.decrypt(value)
    end
  end
end
```

Then it's less automagic and more explicit.